### PR TITLE
Add author placeholder

### LINKED ISSUE #29: Author placeholder/replacement
Hey fregante,

I would like to request to add **author placeholder/replacement of commit**. You could add this placeholder  – `{author}` to `commit-template` where it will "print" author name. You could add it like follows:

```yaml
...
- name: Create release with authors
  uses: fregante/release-with-changelog@v3
  with:
    token: ${{ secrets.GITHUB_TOKEN }}
    commit-template: '- {hash} {title} {author}'
```

And it will produce release with text `- 6e89f5c Update user in examples @fregante`:
- 6e89f5c Update user in examples @fregante

It's better to see it written in changelog or in mail notification, not just after hovering or opening commit.

### PR DESCRIPTION
**Pull request by BelKed**
Resolves #29.

- [x] Add author placeholder
- [x] Update `README.md`
- [x] Add test in `generate-release-notes.test.js`

### CODE REVIEW

**Comment by yakov116** [comment]
@BelKed you should not open a PR from your main (default) branch for a few reasons see [this](https://blog.jasonmeridth.com/posts/do-not-issue-pull-requests-from-your-master-branch/)

You said one yourself yesterday
>@fregante, please, could you merge this pull request now, so I can open another?

_Originally posted by @BelKed in https://github.com/fregante/release-with-changelog/issues/43#issuecomment-803647711_

**Comment by BelKed** [comment]
@yakov116, thanks for pointing on this problem 😄

File: index.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ async function run() {
 			owner,
 			name: releaseTitle.replace('{tag}', pushedTag),
 			tag_name: pushedTag, // eslint-disable-line camelcase
-			body: await generateReleaseNotes({range, exclude, commitTemplate, releaseTemplate, dateFormat, reverseSort}),
+			body: await generateReleaseNotes({octokit, owner, repo, range, exclude, commitTemplate, releaseTemplate, dateFormat, reverseSort}),
```
Comment (fregante) [inline index.js]: The object should now be on multiple lines

Comment (BelKed) [reply index.js]: Like that?
```suggestion
			body: await generateReleaseNotes({
				octokit, owner, repo, range, exclude,
				commitTemplate, releaseTemplate, dateFormat, reverseSort
			}),
```

Comment (BelKed) [reply index.js]: Fixed in 343aa58

**Comment by BelKed** [comment]
@fregante, sorry for long inactivity on this PR.
Anyway, please, could you review?

**Comment by BelKed** [comment]
@fregante, sorry for long inactivity on this PR.
Anyway, please, could you review?

**Comment by fregante** [comment]
CI is not passing

File: index.js [hunk 2]
```diff
@@ -48,7 +48,11 @@ async function run() {
 
 		core.info('Computed range: ' + range);
 
-		const releaseNotes = await generateReleaseNotes({range, exclude, commitTemplate, releaseTemplate, dateFormat, reverseSort, skipOnEmpty});
+		const octokit = getOctokit(core.getInput('token'));
+		const releaseNotes = await generateReleaseNotes({
+			octokit, owner, repo, range, exclude, commitTemplate,
+			releaseTemplate, dateFormat, reverseSort, skipOnEmpty
```
Comment (fregante) [inline index.js]: Can you place these one per line?

**Review by fregante — COMMENTED** [review]
Ensure the tests pass before requesting for review

File: index.js (hunk 2 above)
Comment (BelKed) [reply index.js]: Fixed in 103ae4d

**Comment by fregante** [comment]
Thank you! The big issue is that the token is now required whether one wants the author or not. Also it's silencing a lint rule which should not be silenced. 

While it's interesting I'm not sure it's worth the complexity. Definitely the token cannot be *required* when the authors aren't included.

File: action.yml [hunk 3]
```diff
@@ -52,5 +52,5 @@ outputs:
   skipped:
     description: Will be set to true if skip-on-empty is enabled and a release was not created, false otherwise
 runs:
-  using: 'node16'
+  using: 'node12'
```
Comment (fregante) [inline action.yml:55]: ?

**Review by fregante — CHANGES_REQUESTED** [review]

**Comment by fregante** [comment]
I think I'm gonna avoid this for now, sorry.

File: action.yml (hunk 3 above)
Comment (BelKed) [reply action.yml:55]: `node16` throws an error. Seems like it is not supported by Github Actions:
(I loaded lastet version of this repo – `fregante/release-with-changelog@main`)
```
Error: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''using: node16' is not supported, use 'docker' or 'node12' instead.')
   at GitHub.Runner.Worker.ActionManifestManager.ConvertRuns(IExecutionContext executionContext, TemplateContext templateContext, TemplateToken inputsToken, String fileRelativePath, MappingToken outputs)
   at GitHub.Runner.Worker.ActionManifestManager.Load(IExecutionContext executionContext, String manifestFile)
Error: Fail to load fregante/release-with-changelog/main/action.yml
```

**Pull request closed without merging**